recipe/activate.sh

```diff
@@ -22,18 +22,29 @@ if [ ! -w "$WRAPPER_DIR" ]; then
     return
 fi
 
-# there's only one lib in that folder, but the libname changes
+# Everything that follows amounts to very carefully creating the
+# symlink if need be.
+
+# there's only one lib in the placeholder folder, but the libname changes
 # based on the version so use a loop instead of hardcoding it.
-for f in "$GDB_PREFIX/$PLACEHOLDER/lib/"*.py; do
-    if [ ! -e "$f" ]; then
+for target in "$GDB_PREFIX/$PLACEHOLDER/lib/"*.py; do
+    if [ ! -e "$target" ]; then
         # If the file doesn't exist, skip this iteration of the loop.
         # (This happens when no files are found, in which case the
         # loop runs with f equal to the pattern itself.)
         continue
     fi
-    target="$WRAPPER_DIR/$(basename "$f")"
-    # We have write permissions to WRAPPER_DIR but not necessarily target.
-    # Thus it's safest to delete the target in case it already exists.
-    rm -f "$target"
-    ln -s "$f" "$WRAPPER_DIR/$(basename "$f")"
+    symlink="$WRAPPER_DIR/$(basename "$target")"
+    # Check if symbolic link already exists and points to correct file
+    if [ -L "$symlink" ] && [ "$(readlink "$symlink")" = "$target" ]; then
+        # Stop if it does
+        continue
+    fi
+    # We have write permissions to WRAPPER_DIR but not necessarily the symlink.
+    # Thus rather than editing the symlink, we should create a temporary symlink
+    # and then move it into place.
+    temp_symlink="$(mktemp --tmpdir gdb-symlink-for-arrow-cpp-XXXXX)"
```

Comment:
I had to look up what this does, and it's IMO not obvious (at first I thought it's a directory because of `--tmpdir`).

Also, it seems that OSX doesn't support that option:
```
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix
```

Why not just `-p .`?

Reply:
I wonder if simply calling `ln -f` does the same. Overwrites anything there atomically and changes the owner etc.